Fix bug #39 PWD on windows gives bad path

fabiengb opened the pull request:
Hi,

This is the pull request you requested to fix bug #39, bad path for Windows host.

I have tested this under Windows 7 & RedHat, it seems to work OK.

Fab

sstur commented:
Thanks @fabiengb. These changes look good to me, but they have broken some tests. We need to determine if the tests are wrong or if this is actually a breaking change.

sstur commented:
This appears to break path resolution. There are test cases where a path should resolve relative to `this.root` but instead it resolves relative to `/` which is dangerous.

fabiengb commented:
Hi,
You have a point here.
Can you check the new version of my repo ?

I have implemented actual virtual root for users, preventing them to go upper from the directory they are allowed to.

Plus, i have changed pathEscape function so the path showed to the client is an abstract FTP path and not the or a part of the actual FS path.

impaler commented:
Thanks @fabiengb your changes work for me @sstur would be great to have this in a release for windoze ;)

ikb42 commented:
@sstur will this be merged?  Thanks

sstur commented:
There are some problems with this code. Apart from the stylistic issues (why are there added spaces to the end of lines all over?) I don't think the code will work in a robust way (although we will need some tests to verify this).

sstur commented:
Regarding the style issues, I will add eslint to this project and have it run as part of the tests so we don't have to discuss this on future PRs.

iongion commented:
Can this be applied ?

mk-pmb commented:
I'm confused. I see travis claim success, but the diff view still shows several occurrences of whitespace at end of line.
